From 9fd6977b5d7c29b60cfdb55639c1602b2f44d3ee Mon Sep 17 00:00:00 2001
From: Bruno Martins <bgcngm@gmail.com>
Date: Sun, 20 Feb 2022 23:34:54 +0000
Subject: [PATCH 3/8] p/android-4.4: Enforce eBPF configs

S userspace dropped iptables support and right now only eBPF is
supported.

Change-Id: I2309151cf71216e7e3be01d11f9a40ce04256a2d
---
 p/android-4.4/android-base.config | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/p/android-4.4/android-base.config b/p/android-4.4/android-base.config
index 9db3248..25dd174 100644
--- a/p/android-4.4/android-base.config
+++ b/p/android-4.4/android-base.config
@@ -3,6 +3,7 @@
 # CONFIG_DEVMEM is not set
 # CONFIG_FHANDLE is not set
 # CONFIG_INET_LRO is not set
+# CONFIG_NETFILTER_XT_MATCH_QTAGUID is not set
 # CONFIG_NFSD is not set
 # CONFIG_NFS_FS is not set
 # CONFIG_SYSVIPC is not set
@@ -16,7 +17,9 @@ CONFIG_ASHMEM=y
 CONFIG_AUDIT=y
 CONFIG_BLK_DEV_INITRD=y
 CONFIG_BLOCK=y
+CONFIG_BPF_SYSCALL=y
 CONFIG_CGROUPS=y
+CONFIG_CGROUP_BPF=y
 CONFIG_CGROUP_CPUACCT=y
 CONFIG_CGROUP_FREEZER=y
 CONFIG_CGROUP_SCHED=y
@@ -86,6 +89,7 @@ CONFIG_MODVERSIONS=y
 CONFIG_NET=y
 CONFIG_NETDEVICES=y
 CONFIG_NETFILTER=y
+CONFIG_NETFILTER_XT_MATCH_BPF=y
 CONFIG_NETFILTER_XT_MATCH_COMMENT=y
 CONFIG_NETFILTER_XT_MATCH_CONNLIMIT=y
 CONFIG_NETFILTER_XT_MATCH_CONNMARK=y
@@ -97,9 +101,9 @@ CONFIG_NETFILTER_XT_MATCH_LENGTH=y
 CONFIG_NETFILTER_XT_MATCH_LIMIT=y
 CONFIG_NETFILTER_XT_MATCH_MAC=y
 CONFIG_NETFILTER_XT_MATCH_MARK=y
+CONFIG_NETFILTER_XT_MATCH_OWNER=y
 CONFIG_NETFILTER_XT_MATCH_PKTTYPE=y
 CONFIG_NETFILTER_XT_MATCH_POLICY=y
-CONFIG_NETFILTER_XT_MATCH_QTAGUID=y
 CONFIG_NETFILTER_XT_MATCH_QUOTA2=y
 CONFIG_NETFILTER_XT_MATCH_QUOTA=y
 CONFIG_NETFILTER_XT_MATCH_SOCKET=y
@@ -120,6 +124,7 @@ CONFIG_NETFILTER_XT_TARGET_TCPMSS=y
 CONFIG_NETFILTER_XT_TARGET_TPROXY=y
 CONFIG_NETFILTER_XT_TARGET_TRACE=y
 CONFIG_NET_CLS_ACT=y
+CONFIG_NET_CLS_BPF=y
 CONFIG_NET_CLS_U32=y
 CONFIG_NET_EMATCH=y
 CONFIG_NET_EMATCH_U32=y
-- 
2.53.0.windows.1

